plaid-stl/src/npm/mod.rs

```diff
@@ -0,0 +1,607 @@
+
+use chrono::{DateTime, Utc};
+
+// All the structs which are also used in the Plaid API
+
+/// Permission that is granted to a team over an npm package
+#[derive(Serialize, Deserialize, Clone)]
+pub enum NpmPackagePermission {
+    READ,
+    WRITE,
+}
+
+impl ToString for NpmPackagePermission {
+    fn to_string(&self) -> String {
+        match self {
+            Self::READ => "read".to_string(),
+            Self::WRITE => "write".to_string(),
+        }
+    }
+}
+
+impl NpmPackagePermission {
+    // We do not implement the TryFrom trait because of the error type. We do not want to carry the
+    // NpmError struct here in the STL. Also, we don't care about which error we get from here. If
+    // it fails, the caller will know what to do.
+    pub fn try_from(value: String) -> Result<Self, ()> {
+        match value.as_str() {
+            "read" => Ok(NpmPackagePermission::READ),
+            "write" => Ok(NpmPackagePermission::WRITE),
+            _ => Err(()),
+        }
+    }
+}
+
+/// Teams in the npm organization
+#[derive(Serialize, Deserialize)]
+pub enum NpmTeam {
+    Admins,
+    Developers,
+}
+
+impl ToString for NpmTeam {
+    fn to_string(&self) -> String {
+        match self {
+            Self::Admins => "Admins".to_string(),
+            Self::Developers => "developers".to_string(),
+        }
+    }
+}
+
+/// Role that a user can have in the npm organization
+#[derive(Debug, Deserialize, Serialize)]
+pub enum NpmUserRole {
+    Member,
+    Admin,
+    Owner,
+}
+
+impl NpmUserRole {
+    // We do not implement the TryFrom trait because of the error type. We do not want to carry the
+    // NpmError struct here in the STL. Also, we don't care about which error we get from here. If
+    // it fails, the caller will know what to do.
+    pub fn try_from(value: String) -> Result<Self, ()> {
+        match value.as_str() {
+            "developer" => Ok(NpmUserRole::Member),
+            "admin" => Ok(NpmUserRole::Admin), // untested: I think we do not have Admins
+            "super-admin" => Ok(NpmUserRole::Owner),
+            _ => Err(()),
+        }
+    }
+}
+
+/// A user in the npm organization
+#[derive(Debug, Serialize, Deserialize)]
+pub struct NpmUser {
+    pub username: String,
+    pub role: NpmUserRole,
+}
+
+/// Org-level permission granted to a granular token.
+///
+/// Note: this is incomplete. Other permission levels are possible, but we never needed to add them.
+#[derive(Clone, Serialize, Deserialize)]
+pub enum GranularTokenOrgsPermission {
+    NoAccess,
+}
+
+impl ToString for GranularTokenOrgsPermission {
+    fn to_string(&self) -> String {
+        match self {
+            GranularTokenOrgsPermission::NoAccess => "No access".to_string(),
+        }
+    }
+}
+
+/// Package-level permission granted to a granular token.
+///
+/// Note: this is incomplete. Other permission levels are possible, but we never needed to add them.
+#[derive(Clone, Serialize, Deserialize)]
```

Comment:
Again, would complete this to make it easier to add things later without having to change or update the runtime